src/main/scala/com/microsoft/partnercatalyst/fortis/spark/Pipeline.scala

```diff
@@ -57,7 +57,7 @@ object Pipeline {
       }
 
       def addLanguage(event: ExtendedFortisEvent[T]): ExtendedFortisEvent[T] = {
-        val language = analyzer.detectLanguage(event.details, languageDetector)
+        val language = analyzer.detectLanguage(event.details, new LocalLanguageDetector)
```

Comment:
Since this is on a per-event code path, it'll get constructed for every event, so the lazy vals in the impl won't be reused. Are they light-weight to construct?